File: JsPlc.Ssc.PetrolPricing/JsPlc.Ssc.PetrolPricing.Portal/Scripts/App/sitePricingSettings.js
define(["common", "jquery", "notify"],
    function (common, $, notify) {

        var isLoaded = false,
            error = "",
            settings = {
                MinUnleadedPrice: undefined,
                MaxUnleadedPrice: undefined,
                MinDieselPrice: undefined,
                MaxDieselPrice: undefined,
                MinSuperUnleadedPrice: undefined,
                MaxSuperUnleadedPrice: undefined,
                MinUnleadedPriceChange: undefined,
                MaxUnleadedPriceChange: undefined,
                MinDieselPriceChange: undefined,
                MaxDieselPriceChange: undefined,
                MinSuperUnleadedPriceChange: undefined,
                MaxSuperUnleadedPriceChange: undefined,
                MaxGrocerDriveTimeMinutes: undefined,
                PriceChangeVarianceThreshold: undefined,
                SuperUnleadedMarkupPrice: undefined
            };

        function getSettings() {
            return settings;
        };

        function load(opts) {
            var url = "Sites/SitePricingSettings",
                promise = common.callService("get", url, null);

            promise.done(function (serverData, textStatus, jqXhr) {
                isLoaded = true;
                settings = $.extend({}, serverData);

                if (opts && $.isFunction(opts.success))
                    opts.success(settings);
            })
            .fail(function () {
                isLoaded = false;
                if (opts && $.isFunction(opts.failure))
                    opts.failure();
                console.log('Unable to load Site Pricing Settings');
            });
        };

        // API
        return {
            getSettings: getSettings,
            load: load
        };
    }
);

File: JsPlc.Ssc.PetrolPricing/JsPlc.Ssc.PetrolPricing.Models/SitePricingSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JsPlc.Ssc.PetrolPricing.Models
{
    public class SitePricingSettings
    {
        public double MinUnleadedPrice { get; set; }
        public double MaxUnleadedPrice { get; set; }
        public double MinDieselPrice { get; set; }
        public double MaxDieselPrice { get; set; }
        public double MinSuperUnleadedPrice { get; set; }
        public double MaxSuperUnleadedPrice { get; set; }
        public double MinUnleadedPriceChange { get; set; }
        public double MaxUnleadedPriceChange { get; set; }
        public double MinDieselPriceChange { get; set; }
        public double MaxDieselPriceChange { get; set; }
        public double MinSuperUnleadedPriceChange { get; set; }
        public double MaxSuperUnleadedPriceChange { get; set; }
        public int MaxGrocerDriveTimeMinutes { get; set; }
        public double PriceChangeVarianceThreshold { get; set; }
        public double SuperUnleadedMarkupPrice { get; set; }
        public int DecimalRounding { get; set; }
        public bool EnableSiteEmails { get; set; }
        public string SiteEmailTestAddresses { get; set; }
    }
}
